codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -343,10 +345,107 @@ public void headersWriteShouldOpenStreamForPush() throws Exception {
     }
 
     @Test
+    public void trailersDoNotEndStreamThrows() {
+        writeAllFlowControlledFrames();
+        final int streamId = 6;
+        ChannelPromise promise = newPromise();
```

Comment:
I don't think it is necessary because we are mocking out the underlying transport ... we mostly care when things have been failed (because it is done at the encoder level).